rskj-core/src/test/java/co/rsk/peg/MinimumPegValueTest.java

```diff
@@ -0,0 +1,189 @@
+        new UTXO(getUTXOHash("utxo"), 0, minPeginValue.times(10), 0, false, fed.getP2SHScript()));
+    // the federation wallet, with flyover support
+    Wallet fedWallet = BridgeUtils.getFederationSpendWallet(
+        Context.getOrCreate(networkParameters),
+        fed,
+        utxos,
+        true,
+        mock(BridgeStorageProvider.class));
+    // build release transaction builder for current fed
+    when(activations.isActive(ConsensusRule.RSKIP201)).thenReturn(true);
+    ReleaseTransactionBuilder releaseTransactionBuilder = new ReleaseTransactionBuilder(
+        networkParameters,
+        fedWallet,
+        fed.getAddress(),
+        feePerKb,
+        activations);
+
+    // build batch peg-out transaction
+    ReleaseTransactionBuilder.BuildResult result = releaseTransactionBuilder.buildBatchedPegouts(entries);
+
+    // the proposed feePerKb and peg values are compatible
+    assertEquals(ReleaseTransactionBuilder.Response.SUCCESS, result.getResponseCode());
+  }
+
+  @ParameterizedTest()
+  @MethodSource("providePegMinimumParameters")
+  void checkProposedPegValues_whenTenInputOfMinPeginValueAndTenOutputsOfMinPegoutValue_shouldBuildTransactionSuccessfully(
+      Coin feePerKb, Coin minPeginValue, Coin minPegoutValue) {
+    // build federation
+    Federation fed = new P2shErpFederationBuilder().withNetworkParameters(networkParameters).build();
+    // list of peg-out requests with the given minimum peg-out value
+    List<ReleaseRequestQueue.Entry> entries = new ArrayList<>();
+    for (int i = 0; i < 10; i++) {
+      entries.add(createTestEntry(1000 + i, minPegoutValue));
+    }
+    // list of utxos that contains one utxo with the minimum peg-in value
+    List<UTXO> utxos = new ArrayList<>();
+    for (int i = 0; i < 10; i++) {
+      utxos.add(
+          new UTXO(getUTXOHash("utxo" + i), 0, minPeginValue, 0, false, fed.getP2SHScript()));
+    }
+    // the federation wallet, with flyover support
+    Wallet fedWallet = BridgeUtils.getFederationSpendWallet(
+        Context.getOrCreate(networkParameters),
+        fed,
+        utxos,
+        true,
+        mock(BridgeStorageProvider.class));
+    // build release transaction builder for current fed
+    when(activations.isActive(ConsensusRule.RSKIP201)).thenReturn(true);
+    ReleaseTransactionBuilder releaseTransactionBuilder = new ReleaseTransactionBuilder(
+        networkParameters,
+        fedWallet,
+        fed.getAddress(),
+        feePerKb,
+        activations);
+
+    // build batch peg-out transaction
+    ReleaseTransactionBuilder.BuildResult result = releaseTransactionBuilder.buildBatchedPegouts(entries);
+
+    // the proposed feePerKb and peg values are compatible
+    assertEquals(ReleaseTransactionBuilder.Response.SUCCESS, result.getResponseCode());
+  }
+
+  private static Stream<Arguments> providePegMinimumParameters() {
+    return Stream.of(
+        Arguments.of(Coin.valueOf(24_000L), Coin.valueOf(100_000L), Coin.valueOf(80_000L)));
```

Comment:
Is there a way to get the current feePerKb programatically?

Reply:
Current as in mainnet? You'd have to do a call to the Bridge using a mainnet node. Probably out of scope for a test.

In the code, you can use FeePerKbSupport class to get it

Reply:
I see , thanks